src/usb.c

```diff
@@ -777,37 +777,47 @@ stlink_t *stlink_open_usb(enum ugly_loglevel verbose, bool reset, char serial[16
 
     while (cnt--) {
         libusb_get_device_descriptor( list[cnt], &desc );
-        if (desc.idVendor != STLINK_USB_VID_ST)
-            continue;
-
         if (devBus && devAddr) {
             if ((libusb_get_bus_number(list[cnt]) != devBus)
                 || (libusb_get_device_address(list[cnt]) != devAddr)) {
                 continue;
             }
+            if (desc.idVendor != STLINK_USB_VID_ST) {
+                WLOG("Supported only vendor %04x for STLINK_DEVICE. Choosen device has vendor %04x\n", STLINK_USB_VID_ST, desc.idVendor);
+                goto on_error;
+            }
+        } else {
+            if (desc.idVendor != STLINK_USB_VID_ST)
+                continue;
         }
 
         if ((desc.idProduct == STLINK_USB_PID_STLINK_32L) || (desc.idProduct == STLINK_USB_PID_STLINK_NUCLEO)) {
             struct libusb_device_handle *handle;
 
             ret = libusb_open(list[cnt], &handle);
-            if (ret)
-		continue;
+            if (ret) {
```

Comment:
Better check against legit libusb return code macros instead of != 0. The return code can actual be printed as a string (look it up in the libusb api).